uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -379,9 +397,37 @@ define([
      */
     function initListeners () {
       listeners.push(
-        $rootScope.$on('LeaveRequestPopup::updateBalance', function () {
-          vm.updateBalance();
-          setDaySelectionMode();
+        $rootScope.$on('LeaveRequestPopup::updateAbsenceType', function () {
+          var absenceTypeUnitChanged;
+          var previousAbsenceTypeUnit = vm.selectedAbsenceType.calculation_unit_name;
+
+          vm.selectedAbsenceType = getSelectedAbsenceType();
+          absenceTypeUnitChanged = previousAbsenceTypeUnit !== vm.selectedAbsenceType.calculation_unit_name;
+
+          setOpeningBalance();
+
+          return $q.resolve()
+            .then(absenceTypeUnitChanged && setDaysSelectionMode)
+            .then(function () {
+              if (absenceTypeUnitChanged && isCalculationUnit('hours')) {
+                return loadTimeRangesFromWorkPattern('from');
+              }
+            })
+            .then(setRequestDates)
```

Comment:
what does changing a different absence type have anything to do with setting the dates of the request?

Reply:
If absence type is changed, then "to" date resets because of setDaysSelectionMode().